documentation/technical_writing_slides.md

```diff
@@ -0,0 +1,452 @@
+---
+type: slide
+slideOptions:
+  transition: slide
+  width: 1400
+  height: 900
+  margin: 0.1
+---
+
+<style>
+  .reveal strong {
+    font-weight: bold;
+    color: orange;
+  }
+  .reveal p {
+    text-align: left;
+  }
+  .reveal section h1 {
+    color: orange;
+  }
+  .reveal section h2 {
+    color: orange;
+  }
+  .reveal code {
+    font-family: 'Ubuntu Mono';
+    color: orange;
+  }
+</style>
+
+
+# Technical Writing
+
+---
+
+## 1. Introduction
+
+---
+
+## A Poem
+
+> If people don’t know why your project exists,
+they won’t use it.
+> If people can’t figure out how to install your code,
+they won’t use it.
+> If people can’t figure out how to use your code,
+they won’t use it.
+
+Taken from [Write The Docs](https://www.writethedocs.org/guide/writing/beginners-guide-to-docs/#why-write-docs)
+
+---
+
+Similar message and highly recommended:
+Bangerth and Heister, 2013: [What makes computational open source software libraries successful?](http://dx.doi.org/10.1088/1749-4699/6/1/015010)
+
+---
+
+## Learning Goals
+
+- Raise **awareness** that documentation is crucial in RSE.
+- Understand that there is a difference between documentation and **good documentation**.
+- Get to know the purpose and basic structure of several **standard documentation building blocks** (README, commit message, changelog, ...).
+- **Not**: Learn how to really write good documentation. This takes practice and much more than 90 minutes.
+- **Not**: Any technical tools. This will come next week.
+
+---
+
+## Technical Writer
+
+- ... is a job profile between technical know-how (computer science, engineering, ...) and language.
+- ... can be a role in a software development team (*"documentation"*).
+- There is whole community, e.g. [Write the Docs](https://www.writethedocs.org/).
+
+Much content of this lecture is taken from [Write the Docs](https://www.writethedocs.org/).
+
+---
+
+## Technical Writing in Research Software
+
+- Like in a startup, **no dedicated technical writer in team** (even for large projects). Everybody does everything.
+    - -> You need writing skills.
+- In research, we anyway write a lot (papers, proposals, lecture material, ...).
+    - -> Writing skills are very important anyway.
+
+---
+
+## Why Write Documentation
+
+- You will be using your code in 6 months.
+- You want people to use your code.
+- You want people to help out.
+- You want your code to be better (act of putting words to paper requires a distillation of thought that may not be so easy).
+- You want to be a better writer.
```

Comment:
Same feeling regarding better as above, but here I am not so convinced whether one needs the comparative.

Reply:
I took these from "write the docs", they should know :see_no_evil: